scripts/upgradeable/upgradeable.patch

```diff
@@ -202,12 +137,12 @@ index d94e956a..86bb5713 100644
  import {IERC5267} from "../../interfaces/IERC5267.sol";
  
  /**
-@@ -30,27 +29,19 @@ import {IERC5267} from "../../interfaces/IERC5267.sol";
-  *
-  * _Available since v3.4._
-  *
+@@ -27,28 +26,18 @@ import {IERC5267} from "../../interfaces/IERC5267.sol";
+  * NOTE: In the upgradeable version of this contract, the cached values will correspond to the address, and the domain
+  * separator of the implementation contract. This will cause the `_domainSeparatorV4` function to always rebuild the
+  * separator from the immutable values, which is cheaper than accessing a cached version in cold storage.
+- *
 - * @custom:oz-upgrades-unsafe-allow state-variable-immutable state-variable-assignment
```

Comment:
This part of the patch is removing some natspec comments from the vanilla version.
It looks to me that these natspec comments are only usefull in an upgradeable context. So why do we have them in the vanilla version in the first place ?

Reply:
The vanilla contract is upgradeable-safe even though it uses immutable variables. If we keep these annotations, the transpiler will keep the immutable variables instead of converting them. That said I think the patch removes all immutable variables anyway.

As a side note, I'm realizing that using `unsafe-allow state-variable-assignment` in this way is bad because we only want it to apply to state-variable-immutable and this will allow assignments to any storage variable.

Reply:
> The vanilla contract is upgradeable-safe.

Is that something we want the upgrade plugin to detect ? Using the vanila EIP712 as an implementation for proxy is dangerous.

Reply:
>Using the vanilla EIP712 as an implementation for proxy is dangerous.

Is it? I think the annotation is there because it is not dangerous.

Reply:
You're right. I forgot about the `address(this) == _cachedThis` check. Its indeed safe.